linkerd/proxy/http/src/canonicalize.rs

```diff
@@ -146,8 +141,8 @@ where
                     let target = match ready!(future.poll(cx)) {
                         Ok(Ok(refined)) => {
                             let mut target = original.take().expect("illegal state");
-                            let name = NameAddr::new(refined, target.addr().port());
-                            *target.addr_mut() = name.into();
+                            let name = NameAddr::new(refined, target.as_ref().port());
+                            *target.as_mut() = name.into();
```

Comment:
I guess the one downside of replacing the traits with `AsRef`/`AsMut` is that it's a bit less clear at the callsite what type this is being `AsRef`ed to. 

One thing we _could_ do is keep the trait, but change it to
```rust
pub trait Target {
    fn addr(&self) -> &Addr;
    fn addr_mut(&mut self) -> &mut Addr;
}

impl<T: AsRef<Addr> + AsMut<Addr>> Target for T {
    fn addr(&self) -> &Addr {
        self.as_ref()
    }
    
     fn addr_mut(&mut self) -> &mut Addr {
         self.as_mut()
     }
}
```
and then we could still use methods with semantic names at the callsite, but not require other crates to implement a special trait. 

On the other hand, I'm not sure if this is worth the effort.
```